propnet/ext/matproj.py

```diff
@@ -22,104 +22,70 @@
 
 # list of all available properties
-AVAILABLE_MP_PROPERTIES = list(PROPNET_FROM_MP_NAME_MAPPING.keys()) + ['task_id']
+AVAILABLE_MP_PROPERTIES = list(PROPNET_FROM_MP_NAME_MAPPING.keys()) + ['task_id', 'structure']
 PROPNET_PROPERTIES_ON_MP = list(MP_FROM_PROPNET_NAME_MAPPING.keys())
 
 
-def import_all_props(mp_ids):
-    """Given a list of materials ids, returns a list of all available properties
-    for each material. One list per material id is stored in a list of lists.
-
-    Args:
-      mp_ids: materials ids for which properties are requested.
-
-    Returns:
-      list of lists containing all possible materials properties by mp_id
-
+def import_materials(mp_ids, api_key=None):
     """
-    return import_props(mp_ids, AVAILABLE_MP_PROPERTIES)
-
-
-def import_props(mp_ids, prop_list):
-    """Given a list of materials ids and mp property names returns a list of property
-    instances for each requested material. These lists are stored in an encapsulating
-    list.
-
+    Given a list of material ids, returns a list of Material objects with all
+    available properties from the Materials Project.
     Args:
-      mp_ids: materials ids for which properties are requested.
-      prop_list: requested MP properties for materials.
-
+        mp_ids (list<str>): list of material ids whose information will be retrieved.
+        api_key (str): api key to be used to conduct the query.
     Returns:
-      list of lists containing requested materials properties by mp_id
-
+        (list<Material>): list of material objects with associated data.
     """
-    query = mpr.query(criteria={"task_id": {'$in': mp_ids}}, properties=prop_list)
-    print(query)
-    # properties of all mp-ids inputted; list of lists
-    properties = []
+    if api_key:
+        mpr = MPRester(api_key)
+    else:
+        mpr = MPRester()
+    to_return = []
+    query = mpr.query(criteria={"task_id": {'$in': mp_ids}}, properties=AVAILABLE_MP_PROPERTIES)
     for data in query:
         # properties of one mp-id
-        mat_properties = []
+        mat = Material()
+        iString = data['task_id']
+        mat.add_property(Symbol(SymbolType['structure'], data['structure'], [iString]))
+        mat.add_property(Symbol(SymbolType['lattice_unit_cell'], data['structure'].lattice.matrix, [iString]))
         for key in data:
-            if not data[key] is None:
-                prop_type = _symbol_metadata[PROPNET_FROM_MP_NAME_MAPPING[key]]
-                p = Symbol(prop_type,
-                           data[key],
-                             'Imported from Materials Project',
-                           [(data['task_id'], '')], {})
-                mat_properties.append(p)
-        properties.append(mat_properties)
-    return properties
+            if not data[key] is None and key in PROPNET_FROM_MP_NAME_MAPPING.keys():
+                prop_type = SymbolType[PROPNET_FROM_MP_NAME_MAPPING[key]]
+                p = Symbol(prop_type, data[key], [iString])
```

Comment:
Maybe we should add keyword args to Symbol to make it clearer what this constructor does .... ? not important for this PR, just a thought

Reply:
We can, I don't know if in iPython notebooks / similar you can pull up constructor documentation. Argument types and meanings should be listed there. But, of course, extra clarity is never bad.

We'll have to change the constructor in a lot of places in the code if we make the change. So better earlier than later.

Reply:
> We can, I don't know if in iPython notebooks / similar you can pull up constructor documentation. 

They do, you can either put a ? at the end of the method and it'll pop up documentation, or there's a short cut key to bring it up too.

> We'll have to change the constructor in a lot of places in the code if we make the change. 

We won't. If you specify keyword args in the constructor, but omit them when you call the constructor, it'll still work -- it'll assume the args you provide are in the same order as the keyword args it expects, I believe. Still better to change sooner rather than later, but not a big deal either way as long as argument order isn't changed.